dot/sync/chain_sync.go

```diff
@@ -420,6 +435,24 @@ func (cs *chainSync) sync() {
 				continue
 			}
 
+			worker.retryCount = res.retryCount + 1
+			if worker.retryCount > cs.maxWorkerRetries {
```

Comment:
I'm a little confused on what happens when retryCount has passed the maxWorkerRetries, do we not want to ignore the peer in this case? I'm probably missing some context but im not quite sure what this is doing

Reply:
I guess what Im asking is the log says the peer is being discarded, so how/where is that happening?

Reply:
the worker isn't tied to a specific peer, in this case what's happening is that the worker has tried to sync with various peers `maxWorkerRetries` times, and we want to just cancel it since it's been retried too much. I saw a lot of infinite looping without this